Fixed bind command, and added rough vtx implementation

SpektrumRC opened the pull request.

DieHertz committed:
Added SRXLv2 implementation

MiguelFAlvarez committed:
Fixed bind command, and added rough vtx implementation

MiguelFAlvarez committed:
Changed initialization of bus_master_device_id to 0xFF. This means that when the flight controller reboots, but the rx is already sending data, sending 0xFF as bus master id will make the receiver request a handshake again.
Fixed baud rate toggling when invalid packets are received. Initially, re-connection wouldn't work on 400k baud.

mikeller committed:
More ACC_x_ALIGN cleanup (#8287)

More ACC_x_ALIGN cleanup

mikeller committed:
Enabled Multi Gyro on OmnibusF4 Nano V7 (#8267)

Enabled Multi Gyro on OmnibusF4 Nano V7

mikeller committed:
Remove deprecated USE_MULTI_GYRO (#8281)

Remove deprecated USE_MULTI_GYRO

mikeller committed:
Increase async FAT filesystem cache sectors to prevent blackbox header corruption (#8284)

Increase async FAT filesystem cache sectors to prevent blackbox header corruption

mikeller committed:
Prevent 'system_hse_mhz' from showing in non-F4 'diff'/'dump'. (#8256)

Prevent 'system_hse_mhz' from showing in non-F4 'diff'/'dump'.

DieHertz commented on src/main/rx/srxlv2.c:
```diff
@@ -205,7 +208,25 @@ bool srxlv2ProcessControlData(const srxlv2ControlDataSubHeader* control_data, rx
+                                    ((vtxData->region & 0x01) << 3) |
+                                    ((vtxData->power & 0x07));
+            #if defined(USE_SPEKTRUM_VTX_CONTROL) && defined(USE_VTX_COMMON)
+            spektrumHandleVtxControl(vtxControl);
```
That's a good solution, reusing functions makes in more robust in this case.

mikeller committed:
Really fixed flash erasing. (#8300)

Really fixed flash erasing.

DieHertz commented on src/main/rx/srxlv2.c:
```diff
@@ -657,6 +691,8 @@ void srxlv2Bind(void)
         .payload = {
             .request = EnterBindMode,
             .device_id = bus_master_device_id,
+            .bind_type = DMSX_11ms,
```
This was something that puzzled me. Should the bind type hard-coded or we should allow specifying it in CLI?
At first I thought these fields are only used in the GetBindInfo request, and filled by the receiver.

SpektrumRC replied:
That's a good point. I think it should be a CLI parameter that defaults to DSMX_11ms, and the "srxlv2_bind" cli command uses the value from that parameter. That way, it should still use this value by default unless the user specifies otherwise. Currently, there should be no need to use anything other than DSMX_11ms for the receivers that will currently support srxlv2.
These fields are used by both requests. For example in a multi receiver system, the FC can GetBindInfo from one receiver, and then send that info to the other receivers so they match the bind state of the initial one, without having to rebind each one.

DieHertz commented on src/main/rx/srxlv2_types.h:
```diff
@@ -92,7 +92,7 @@ typedef enum {
 
 typedef enum {
-    NotBound = 0xAE,
+    NotBound = 0x0,
```
Where did `0xAE` came from? Did I misinterpret it?

SpektrumRC replied:
You had it right based on the documentation, but this was one of the things Matt had to change when fixing the whole bind command implementation. I believe the change was related to some logic required to handle multi receiver systems.

mikeller committed:
Merge pull request #8302 from mikeller/fix_uint32_t_printing

Fixed setting / printing of VAR_UINT32 values.

mikeller committed:
Fixed printing of reset DMA options in 'diff'. (#8303)

Fixed printing of reset DMA options in 'diff'.

mikeller committed:
Add timer info to CLI 'timer' command. (#8305)

Add timer info to CLI 'timer' command.

mikeller committed:
Added selection by alternate function to CLI 'timer'. (#8306)

Added selection by alternate function to CLI 'timer'.

mikeller committed:
Fix VTX chan to 0 via SetFreqByMHz (#8338)

Fix VTX chan to 0 via SetFreqByMHz

mikeller committed:
SPRacingF3NEO - Add BARO support. (#8321)

SPRacingF3NEO - Add BARO support.

mikeller committed:
Merge pull request #8328 from solarhacksaw/fix_airbotf7_adc_pins

fix adc pins

mikeller committed:
Fix copy of profiles with last index (#8331)

Fix copy of profiles with last index

mikeller committed:
TARGET Update: add IIC Baro, SoftSerial, 1G Flash support for EXF722DUAL board. (#8333)

TARGET Update: add IIC Baro, SoftSerial, 1G Flash support for EXF722DUAL board.

mikeller committed:
SPARKY2 Assign correct pins for I2C1 (#8346)

SPARKY2 Assign correct pins for I2C1

mikeller committed:
Fix flash m25p16 driver sector erase (#8342)

Fix flash m25p16 driver sector erase

mikeller committed:
Changed 'timer' output to list timers based on alternate function. (#8320)

Changed 'timer' output to list timers based on alternate function.

mikeller committed:
Merge pull request #8356 from IllusionFpv/gps-new-descent-distance-fix

Add minimum descent distance

mikeller committed:
[VIVAF4AIO] Fix for non-working ACC (#8349)

[VIVAF4AIO] Fix for non-working ACC

MiguelFAlvarez committed:
Commented unused EXTI init
